src/test/java/com/salesforce/storm/spout/dynamic/FactoryManagerTest.java

```diff
@@ -55,30 +52,26 @@
 /**
  * Test that {@link FactoryManager} creates instances correctly.
  */
-@RunWith(DataProviderRunner.class)
 public class FactoryManagerTest {
 
-    @Rule
-    public ExpectedException expectedExceptionCreateNewFailedMsgRetryManagerInstanceMissingConfig = ExpectedException.none();
-
     /**
      * Tests that if you fail to pass a deserializer config it throws an exception.
      */
     @Test
     public void testCreateNewFailedMsgRetryManagerInstanceMissingConfig() {
-        // Try with UTF8 String deserializer
-        final Map config = Maps.newHashMap();
-        final FactoryManager factoryManager = new FactoryManager(config);
-
-        // We expect this to throw an exception.
-        expectedExceptionCreateNewFailedMsgRetryManagerInstanceMissingConfig.expect(IllegalStateException.class);
-        factoryManager.createNewFailedMsgRetryManagerInstance();
+        Assertions.assertThrows(IllegalStateException.class, () -> {
+            // Try with UTF8 String deserializer
+            final Map config = Maps.newHashMap();
```

Comment:
Probably should move this up and out of the code block.  Otherwise this could falsely trigger on the constructor instead of the createNewFialedMsgRetryManagerInstance() call.